packages/core/test/algorithm.test.ts

```diff
@@ -86,7 +86,6 @@ describe('algorithm', () => {
       'a',
       'b1',
       'c1',
-      'c2',
```

Comment:
I've changed the implementation of the `getConfiguration` slightly and this was the only test that has failed in the result. As mentioned in another comment - I don't know why `getStateNodes` accepts `prevNodes`. What's the exact purpose of `getConfiguration`? I use it to resolve initial nodes & ancestors of input nodes.

Reply:
@davidkpiano would love your commentary on that

Reply:
The previous state nodes are needed if only a partial configuration was given, e.g., if a compound node was given and not the child nodes, we can determine the child nodes from the previous config.

Reply:
So is this intended as the public API? Because as far as I can tell such a situation never happens internally.

Reply:
Not public API. Can safely be removed.